projects/storefrontlib/src/cms-structure/services/feature-modules.service.ts

```diff
@@ -0,0 +1,254 @@
+import {
+  Compiler,
+  Injectable,
+  InjectFlags,
+  Injector,
+  NgModuleFactory,
+  NgModuleRef,
+  OnDestroy,
+} from '@angular/core';
+import {
+  CMSComponentConfig,
+  CmsComponentMapping,
+  CmsConfig,
+  ConfigChunk,
+  ConfigInitializerService,
+  configurationFactory,
+  DefaultConfigChunk,
+  FeatureModuleConfig,
+} from '@spartacus/core';
+import {
+  combineLatest,
+  defer,
+  forkJoin,
+  from,
+  Observable,
+  of,
+  queueScheduler,
+} from 'rxjs';
+import { map, observeOn, pluck, share, switchMap, tap } from 'rxjs/operators';
+
+interface FeatureInstance extends FeatureModuleConfig {
+  moduleRef?: NgModuleRef<any>;
+  depsModules?: any[];
+  injectors?: Injector[];
+  componentsMappings?: CMSComponentConfig;
+}
+
+/**
+ * Service responsible for resolving feature domains
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class FeatureModulesService implements OnDestroy {
+  // maps componentType to feature
+  private componentFeatureMap: Map<string, string> = new Map();
+
+  private featureModulesConfig?: {
+    [featureName: string]: FeatureInstance;
+  };
+
+  private featureResolvers: Map<
+    string,
+    Observable<FeatureInstance>
+  > = new Map();
+
+  private features: Map<string, FeatureInstance> = new Map();
+
+  private dependencyModules = new Map<any, NgModuleRef<any>>();
+
+  constructor(
+    protected configInitializer: ConfigInitializerService,
+    protected compiler?: Compiler,
+    protected injector?: Injector
```

Comment:
Minor: those constructor dependencies can be required. 

Note: we should remove them from cms-components.service (they are not used there)

Reply:
👍